packages/data/addon/utils/classes/cache.ts

```diff
@@ -83,12 +80,13 @@ export default class Cache<T> {
    * @returns {boolean} boolean indicating presence of given key in cache
    */
   has(cacheKey: string): boolean {
+    this._removeStaleItems();
```

Comment:
Do we need to clean the entire cache for a lookup? Is it possible to just remove the single entry if it is stale?